### PR TITLE
Modify 'web suite' link to be a dropdown menu containing apps

### PR DESCRIPTION
**Pull request by VoxSecundus**
This PR introduces a change to the landing page navbar. The conditionally used `Web Suite` button is now a split dropdown button. The old button functionality remains the same, but it is now followed by a dropdown button listing the available webapps.

### CODE REVIEW

**Review by benarmston — CHANGES_REQUESTED** [review]
Code-wise this looks right.  I have some styling comments to make and I'm going to expand the scope of this work too.

Having the items in the menu bold doesn't look quite right.  In particular, it doesn't allow to tell which item I'm hovering over.

image

I think these would look better with the app icon included too.

image

And for the increase in scope, also please include the icon in the sidebar too.

image

**Comment by VoxSecundus** [comment]
Before I replicate the updated dropdown in the webapp components repo, do these look alright? The icons are set to be fixed width so that the app titles line up.
Screenshot_2022-04-05_17-28-19

Screenshot_2022-04-05_17-28-12

File: landing-page/types/headnode/layouts/navbar.html.erb [hunk 1]
```diff
@@ -62,15 +62,38 @@
...
-                  >
-                      <%=
-                        branding('brandbar.apps_link.text') ||
-                          'Web Suite'
-                        %>
-                  </a>
+                  <div class="btn-group">
+                      <a
+                          class="nav-link nav-menu-button"
+                          href="<%= prefix_url("/apps") %>/apps"
```
Comment (benarmston) [inline landing-page/types/headnode/layouts/navbar.html.erb:68]: We don't need both `/apps` here.

File: landing-page/types/headnode/layouts/sidebar.html.erb [hunk 2]
```diff
@@ -36,10 +36,12 @@
       <h2>Quick Access</h2>
       <ul class="w-100">
         <% app_items.each do |app| %>
+          <% icon = "<span class=\"fa fa-solid fa-fw fa-#{app[:fa_icon]}\"></span>" %>
+          <% title = "<span class=\"title\">#{app[:short_title] || app[:title]}></span>" %>
```
Comment (benarmston) [inline landing-page/types/headnode/layouts/sidebar.html.erb:40]: There's an extra `>` on this line (or somewhere).

image

**Review by benarmston — CHANGES_REQUESTED** [review]
The icons are now fixed.  I've found a couple more issues for you to fix.  Could you also take a look at the styling when the alces branding is enabled.  The dropdown arrow isn't visible; the menu is escaping off the edge of the browser and the hover styling obscures the menu item.

image

**Comment by VoxSecundus** [comment]
@benarmston As a suggestion to fix both the menu going off screen and the issue regarding menu alignment in `flight-webapp-components`, do you take any issue with having the menu be anchored to the bottom right of the menu instead of the bottom left?

**Comment by benarmston** [comment]
> @benarmston As a suggestion to fix both the menu going off screen and the issue regarding menu alignment in `flight-webapp-components`, do you take any issue with having the menu be anchored to the bottom right of the menu instead of the bottom left?

Not immediately.  Give it a go and I'll see what it looks like.

**Comment by benarmston** [comment]
I've pushed some changes making minor adjustments to the split toggle arrow padding.  Other than that this is looking good.  Ready to merge when the web app components work is.

File: landing-page/types/alces-branding/content/styles/branding.css [hunk 3]
```diff
@@ -18,9 +18,17 @@ body {
 .navbar .dropdown-menu {
     background-color: rgba(0, 0, 0, 0.9);
 }
```
Comment (benarmston) [inline landing-page/types/alces-branding/content/styles/branding.css:20]: I think we need a corresponding rule for `.navbar .dropdown-item` to set the colour to white.  That should fix the webapp issue dropdown item colour issue.

File: landing-page/types/alces-branding/content/styles/branding.css [hunk 4]
```diff
@@ -18,9 +18,17 @@ body {
...
+.navbar .dropdown-item:hover span {
+    color: black;
+}
 .navbar a.nav.dropdown-item:hover {
     background-color: #586775;
 }
```
Comment (benarmston) [inline landing-page/types/alces-branding/content/styles/branding.css:26]: The second of these rules applies the hover styling to the account menu.  Can you remove the `.navbar .dropdown-item:hover span` rule and make the `.navbar a.nav.dropdown-item:hover` rule also match the dropdown's in the new dropdown menu.

Perhaps make the selector more generic or perhaps add a second specific selector to the rule.

File: landing-page/types/alces-branding/content/styles/branding.css [hunk 5]
```diff
@@ -32,31 +40,43 @@ body {
...
     font-weight: normal;
-    border-right-width: 1px;
-    border-right-style: dotted;
-    border-right-color: rgb(88, 103, 117);
     padding: 8px 24px 8px 24px;
 }
+.navbar .navbar-nav .dropdown-item.nav-link {
+    padding-left: 8px;
+    padding-right: 8px;
+}
```
Comment (benarmston) [inline landing-page/types/alces-branding/content/styles/branding.css:69]: I think the default padding-{left,right} of 24px is probably better.  Can you remove this please?

File: landing-page/types/headnode/layouts/navbar.html.erb [hunk 6]
```diff
@@ -61,16 +61,39 @@
...
+                          aria-haspopup="true"
+                          aria-expanded="false"
+                      >
+                          <span class="sr-only">Toggle Dropdown</span>
+                      </a>
+                      <div class="dropdown-menu dropdown-menu-right">
+                          <% app_items.each do |app| %>
+                              <a
+                                  href="<%= app[:path] %>"
+                                  class="dropdown-item nav-link nav-menu-button"
```
Comment (benarmston) [inline landing-page/types/headnode/layouts/navbar.html.erb:89]: Can you remove the `nav-link` and `nav-menu-button` classes.  The bootstrap documentation doesn't add `nav-link` to its nav dropdown items (https://getbootstrap.com/docs/4.6/components/navs/#pills-with-dropdowns).  I don't think we need to either.

**Review by benarmston — CHANGES_REQUESTED** [review]
After following through the flight direct branding dropdown item colour issues, I have some more comments on this PR.  These comments go hand-in-hand with those I've added to https://github.com/openflighthpc/flight-webapp-components/pull/7.

**Comment by benarmston** [comment]
@VoxSecundus please be aware I added some commits to this PR when I thought it was good.  Please don't lose them.

**Pull request merged**